src/main/java/ch/njol/skript/expressions/ExprSeaPickles.java

```diff
@@ -0,0 +1,153 @@
+
+@Name("Sea Pickles")
+@Description("An expression to obtain or modify data relating to the pickles of a sea pickle block.")
+@Examples({
+	"on block break:",
+	"\ttype of block is sea pickle",
+	"\tsend \"Wow! This stack of sea pickles contained %event-block's sea pickle count% pickles!\"",
+	"\tsend \"It could've contained a maximum of %event-block's maximum sea pickle count% pickles!\"",
+	"\tsend \"It had to have contained at least %event-block's minimum sea pickle count% pickles!\"",
+	"\tcancel event",
+	"\tset event-block's sea pickle count to event-block's maximum sea pickle count",
+	"\tsend \"This bad boy is going to hold so many pickles now!!\""
+})
+@Since("INSERT VERSION")
+public class ExprSeaPickles extends SimplePropertyExpression<Block, Long> {
+
+	static {
+		register(ExprSeaPickles.class, Long.class, "[:(min|max)[imum]] [sea] pickle(s| (count|amount))", "blocks");
+	}
+
+	private boolean minimum, maximum;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		minimum = parseResult.hasTag("min");
+		maximum = parseResult.hasTag("max");
+		return super.init(exprs, matchedPattern, isDelayed, parseResult);
+	}
+
+	@Override
+	@Nullable
+	public Long convert(Block block) {
+		BlockData blockData = block.getBlockData();
+		if (!(blockData instanceof SeaPickle))
+			return null;
+
+		SeaPickle pickleData = (SeaPickle) blockData;
+
+		if (maximum)
+			return (long) pickleData.getMaximumPickles();
+		if (minimum)
+			return (long) pickleData.getMinimumPickles();
+		return (long) pickleData.getPickles();
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		if (minimum || maximum)
+			return null;
+		switch (mode) {
+			case SET:
+			case ADD:
+			case REMOVE:
+			case RESET:
+			case DELETE:
+				return CollectionUtils.array(Number.class);
+			default:
+				return null;
+		}
+	}
+
+	@Override
+	public void change(Event event, Object @Nullable [] delta, ChangeMode mode) {
+		if (delta == null && mode != ChangeMode.RESET && mode != ChangeMode.DELETE)
+			return;
+
+		int change = delta != null ? ((Number) delta[0]).intValue() : 0;
+		if (mode == ChangeMode.REMOVE)
+			change *= -1;
+
+		for (Block block : getExpr().getArray(event)) {
+
+			// Obtain pickle data
+			BlockData blockData = block.getBlockData();
+			if (!(blockData instanceof SeaPickle))
+				return;
+			SeaPickle pickleData = (SeaPickle) blockData;
+
+			int newPickles = change;
+
+			// Calculate new pickles value
+			switch (mode) {
+				case ADD:
+				case REMOVE:
+					newPickles += pickleData.getPickles();
+				case SET:
+					if (newPickles != 0) { // 0 = delete pickles
+						newPickles = Math.max(pickleData.getMinimumPickles(), newPickles); // Ensure value isn't too low
+						newPickles = Math.min(pickleData.getMaximumPickles(), newPickles); // Ensure value isn't too high
+					}
+					break;
+				case RESET:
+					newPickles = pickleData.getMaximumPickles();
```

Comment:
Why does resetting the pickles set it to the maximum instead of the minimum?

Reply:
oops